server/forms.py

```diff
@@ -33,6 +53,38 @@ class MultiCheckboxField(SelectMultipleField):
     widget = widgets.ListWidget(prefix_label=False)
     option_widget = widgets.CheckboxInput()
 
+class BackupUploadField(FileField):
+    def upload_backup_files(self, backup):
+        """Update a Backup's attributes based on form contents. If successful,
+        return True; otherwise, add errors to the form and return False.
+        """
+        assignment = backup.assignment
+        templates = assignment.files
+
+        files = {}
+        for upload in request.files.getlist(self.name):
+            data = upload.read()
+            if len(data) > 2 * 1024 * 1024:  # file is too large (over 2 MB)
+                self.errors.append(
+                    '{} is larger than the maximum file size '
+                    'of 2MB'.format(upload.filename))
+                return False
+            files[upload.filename] = str(data, 'utf-8')
```

Comment:
To prevent 500s can we have a try catch? 
```
try: 
  files[upload.filename] = str(data, 'utf-8')
except UnicodeDecodeError
  flash("Could not upload {}".format(upload.filename), 'warning') 
```